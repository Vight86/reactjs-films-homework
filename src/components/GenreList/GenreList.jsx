import React from 'react';
import propTypes from 'prop-types';
import styles from './GenreList.scss';

const GenreList = ({ className, order, children }) => (
  <ul className={styles[className]} style={{ order }}>
    {children.map((item) => <li key={item}>{item}</li>)}
  </ul>
);

GenreList.propTypes = {
  className: propTypes.string.isRequired,
  order: propTypes.number,
  children: propTypes.arrayOf(propTypes.string).isRequired,
};

GenreList.defaultProps = {
  order: null,

};

export default GenreList;
