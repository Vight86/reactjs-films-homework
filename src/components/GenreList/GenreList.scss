@import '../../utils/variables';

.genreList {
  margin: 0;
}

.secondary {
  @extend .genreList;

  color: $color-tertiary;
  font-size: 2.2em;
  font-weight: 500;
}
