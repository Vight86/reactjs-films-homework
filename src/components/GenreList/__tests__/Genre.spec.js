import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import GenreList from '../index';

let renderer;
let result;

beforeEach(() => {
  renderer = new ShallowRenderer();
  renderer.render(
    <GenreList className="test">
      {['test', 'test1']}
    </GenreList>,
  );
  result = renderer.getRenderOutput();
});

afterEach(() => {
  renderer = null;
  result = null;
});

describe('render GenreList component', () => {
  it('render correctly', () => {
    expect(result).toMatchSnapshot();
  });
});
