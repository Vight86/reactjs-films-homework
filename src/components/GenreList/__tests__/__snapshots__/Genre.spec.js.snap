// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render GenreList component render correctly 1`] = `
<ul
  className="test"
  style={
    Object {
      "order": null,
    }
  }
>
  <li>
    test
  </li>
  <li>
    test1
  </li>
</ul>
`;
